core/base/signedDistanceField/SignedDistanceField.h

```diff
@@ -0,0 +1,720 @@
+      }
+    }
+  }
+  // Can not be parallelized because some vertices can appear more than once
+  for(unsigned int i = 0; i < verticesIntersected.size(); ++i) {
+    for(auto &vertexId : verticesIntersected[i]) {
+      vertexIntersection[vertexId] = true;
+      edgeCrossing[vertexId] = 1;
+    }
+  }
+  this->printMsg("Find intersection edges", 1.0, t_intersect.getElapsedTime(),
+                 this->threadNumber_);
+
+  // ===================================
+  // === Find Outside vertices
+  // ===================================
+  Timer t_outside;
+  // We search for an outside vertex
+  SimplexId outsideVertex = std::numeric_limits<SimplexId>::max();
+  for(unsigned int i = 0; i < vertexIntersection.size(); i++) {
+    if(not vertexIntersection[i]) {
+      outsideVertex = i;
+      std::stringstream ss;
+      ss << "First outside vertex found : " << outsideVertex;
+      printMsg(ss.str());
+      break;
+    }
+  }
+  findOutsideVertices(
+    outsideVertex, englobanteTriangulation, vertexIntersection, isInterior);
+  this->printMsg("Find outside vertices", 1.0, t_outside.getElapsedTime(),
+                 this->threadNumber_);
+
+  // ======================================================
+  // === Signed Distance Field
+  // ======================================================
+  // Find the distance between each point of the grid and the nearest point in
+  // the domain
+  Timer t_distance;
+  printMsg("Start distance computation...");
+
+  std::vector<int> verticesCrossing;
+  if(backend_ != 0) {
+    for(unsigned int i = 0; i < vertexIntersection.size(); ++i)
+      if(vertexIntersection[i])
+        verticesCrossing.emplace_back(i);
+    noVertices = verticesCrossing.size();
+  }
+
+#ifdef TTK_ENABLE_OPENMP
+#pragma omp parallel for num_threads(threadNumber_)
+#endif
+  for(int i = 0; i < noVertices; i++) {
+    auto vertexId = i;
+    if(backend_ != 0)
+      vertexId = verticesCrossing[i];
+
+    ttk::SurfaceGeometrySmoother smoother;
+
+    float x = 0, y = 0, z = 0;
+    englobanteTriangulation->getVertexPoint(vertexId, x, y, z);
+
+    ttk::SurfaceGeometrySmoother::Point point;
+    point[0] = x;
+    point[1] = y;
+    point[2] = z;
+
+    std::vector<float> dists(triangulation->getNumberOfVertices());
+    ttk::SurfaceGeometrySmoother::ProjectionInput projectionInput{
+      static_cast<size_t>(vertexId), point,
+      smoother.getNearestSurfaceVertex(point, dists, *triangulation)};
+
+    std::vector<SimplexId> visitedTriangles{};
+    std::vector<bool> trianglesTested(
+      triangulation->getNumberOfTriangles(), false);
+
+    ttk::VisitedMask vm{trianglesTested, visitedTriangles};
+    std::vector<float> dists2(triangulation->getNumberOfVertices());
+    std::stack<ttk::SimplexId> trianglesToTest;
+    bool reverseProjection = false;
+
+    ttk::SurfaceGeometrySmoother::ProjectionResult projectionResult
+      = smoother.findProjection(projectionInput, vm, dists2, trianglesToTest,
```

Comment:
this function may need to be moved to the geometry class too.

Reply:
Done in the next commit!